faiss/gpu/impl/IVFBase.cu

```diff
@@ -389,6 +389,9 @@ void IVFBase::addEncodedVectorsToList_(
     // Handle the indices as well
     addIndicesFromCpu_(listId, indices, numVecs);
 
+    // save index mapping
+    indexMapping(listId, indices, numVecs);
```

Comment:
whether or not the index mapping is maintained on the host should be an optional feature for a `GpuIndexIVFPQ` index, which is initialized upon index construction (e.g., in `GpuIndexIVFPQConfig`)